src/pylint_odoo/checkers/import_checker.py

```diff
@@ -0,0 +1,128 @@
+import re
+from typing import Union
+
+from astroid import Import, ImportFrom, Module
+from astroid.modutils import modpath_from_file
+from pylint.checkers.utils import only_required_for_messages
+from pylint.lint import PyLinter
+
+from .odoo_base_checker import OdooBaseChecker
+
+ODOO_MSGS = {
+    # C->convention R->refactor W->warning E->error F->fatal
+    "E8401": (
+        "This python module is not imported and has no effect",
+        "unused-module",
+        "",
+    ),
+}
+
+EXCLUDED = re.compile("__manifest__.py$|__openerp__.py$|__init__.py$|tests/|migrations/|tests\\\\|migrations\\\\")
+
+
+class ImportChecker(OdooBaseChecker):
+    msgs = ODOO_MSGS
+    name = "odoolint"
+
+    def __init__(self, linter: PyLinter):
+        self.imports = set()
+        self.modules = set()
+
+        super().__init__(linter)
+
+    @staticmethod
+    def get_module_from_node(node, max_attempts: int = 15) -> Union[None, Module]:
+        """Get the module a node belongs to.
```

Comment:
Fix semantic, this is not clear

Did you mean "Return whether the module "a" belongs to the provided node"?

I'm not sure